tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs

```diff
@@ -215,6 +215,25 @@ public void FlushAsyncCompletedAfterPreCancellation()
         }
 
         [Fact]
+        public void FlushAsyncReturnsIsCancelOnCancelPendingFlushBeforeGetResult()
+        {
+            var writableBuffer = Pipe.Writer.Alloc(MaximumSizeHigh);
+            writableBuffer.Advance(MaximumSizeHigh);
+            var awaitable = writableBuffer.FlushAsync();
+
+            Assert.False(awaitable.IsCompleted);
+            awaitable.OnCompleted(() => {});
+
+            Pipe.Reader.Advance(Pipe.Reader.ReadAsync().GetResult().Buffer.End);
+            Pipe.Writer.CancelPendingFlush();
```

Comment:
I think it makes sense to test flipping the order of the above two lines to verify Advance won't clobber the canceled state.